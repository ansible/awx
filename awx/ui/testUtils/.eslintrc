{
  "rules": {
    "react/jsx-pascal-case": 0
  }
}
