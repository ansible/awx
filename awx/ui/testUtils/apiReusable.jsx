// eslint-disable-next-line import/prefer-default-export
export function describeNotificationMixin (Model, name) {
  describe(name, () => {

    let mockHttp;
    let ModelAPI;
    beforeEach(() => {
      mockHttp = ({ post: jest.fn(() => Promise.resolve()) });
      ModelAPI = new Model(mockHttp);
    })

    afterEach(() => {
      jest.resetAllMocks();
    });

    const parameters = ['success', 'error'];

    parameters.forEach((type) => {
      const label = `[notificationType=${type}, associationState=true`;
      const testName = `associateNotificationTemplate ${label} makes expected http calls`;

      test(testName, async () => {
        await ModelAPI.associateNotificationTemplate(1, 21, type);

        const expectedPath = `${ModelAPI.baseUrl}1/notification_templates_${type}/`;
        expect(mockHttp.post).toHaveBeenCalledTimes(1);

        const expectedParams = { id: 21 };
        expect(mockHttp.post.mock.calls.pop()).toEqual([expectedPath, expectedParams]);

      });
    });

    parameters.forEach((type) => {
      const label = `[notificationType=${type}, associationState=false`;
      const testName = `disassociateNotificationTemplate ${label} makes expected http calls`;

      test(testName, async () => {
        await ModelAPI.disassociateNotificationTemplate(1, 21, type);

        const expectedPath = `${ModelAPI.baseUrl}1/notification_templates_${type}/`;
        expect(mockHttp.post).toHaveBeenCalledTimes(1);

        const expectedParams = { id: 21, disassociate: true };
        expect(mockHttp.post.mock.calls.pop()).toEqual([expectedPath, expectedParams]);

      });
    });
  });
}
