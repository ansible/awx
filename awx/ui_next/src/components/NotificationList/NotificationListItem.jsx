import React from 'react';
import { shape, number, string, bool, func } from 'prop-types';
import { withI18n } from '@lingui/react';
import { t } from '@lingui/macro';
import { Link } from 'react-router-dom';
import { Switch } from '@patternfly/react-core';
import { Tr, Td } from '@patternfly/react-table';
import { ActionsTd, ActionItem } from '../PaginatedTable';

function NotificationListItem({
  canToggleNotifications,
  notification,
  detailUrl,
  approvalsTurnedOn,
  startedTurnedOn,
  successTurnedOn,
  errorTurnedOn,
  toggleNotification,
  i18n,
  typeLabels,
  showApprovalsToggle,
}) {
  return (
    <Tr id={`notification-row-${notification.id}`}>
      <Td id={`notification-${notification.id}`} dataLabel={i18n._(t`Name`)}>
        <Link to={`${detailUrl}`}>
          <b>{notification.name}</b>
        </Link>
      </Td>
      <Td dataLabel={i18n._(t`Type`)}>
        {typeLabels[notification.notification_type]}
      </Td>
      <ActionsTd
        dataLabel={i18n._(t`Options`)}
        gridColumns="120px 120px 120px 120px"
      >
        <ActionItem visible={showApprovalsToggle}>
          <Switch
            id={`notification-${notification.id}-approvals-toggle`}
            label={i18n._(t`Approval`)}
            labelOff={i18n._(t`Approval`)}
            isChecked={approvalsTurnedOn}
            isDisabled={!canToggleNotifications}
            onChange={() =>
              toggleNotification(
                notification.id,
                approvalsTurnedOn,
                'approvals'
              )
            }
            aria-label={i18n._(t`Toggle notification approvals`)}
          />
        </ActionItem>
        <ActionItem visible>
          <Switch
            id={`notification-${notification.id}-started-toggle`}
            label={i18n._(t`Start`)}
            labelOff={i18n._(t`Start`)}
            isChecked={startedTurnedOn}
            isDisabled={!canToggleNotifications}
            onChange={() =>
              toggleNotification(notification.id, startedTurnedOn, 'started')
            }
            aria-label={i18n._(t`Toggle notification start`)}
          />
        </ActionItem>
        <ActionItem visible>
          <Switch
            id={`notification-${notification.id}-success-toggle`}
            label={i18n._(t`Success`)}
            labelOff={i18n._(t`Success`)}
            isChecked={successTurnedOn}
            isDisabled={!canToggleNotifications}
            onChange={() =>
              toggleNotification(notification.id, successTurnedOn, 'success')
            }
            aria-label={i18n._(t`Toggle notification success`)}
          />
        </ActionItem>
        <ActionItem visible>
          <Switch
            id={`notification-${notification.id}-error-toggle`}
            label={i18n._(t`Failure`)}
            labelOff={i18n._(t`Failure`)}
            isChecked={errorTurnedOn}
            isDisabled={!canToggleNotifications}
            onChange={() =>
              toggleNotification(notification.id, errorTurnedOn, 'error')
            }
            aria-label={i18n._(t`Toggle notification failure`)}
          />
        </ActionItem>
      </ActionsTd>
    </Tr>
  );
}

NotificationListItem.propTypes = {
  notification: shape({
    id: number.isRequired,
    name: string.isRequired,
    notification_type: string.isRequired,
  }).isRequired,
  canToggleNotifications: bool.isRequired,
  detailUrl: string.isRequired,
  approvalsTurnedOn: bool,
  errorTurnedOn: bool,
  startedTurnedOn: bool,
  successTurnedOn: bool,
  toggleNotification: func.isRequired,
  typeLabels: shape().isRequired,
  showApprovalsToggle: bool,
};

NotificationListItem.defaultProps = {
  approvalsTurnedOn: false,
  errorTurnedOn: false,
  startedTurnedOn: false,
  successTurnedOn: false,
  showApprovalsToggle: false,
};

export default withI18n()(NotificationListItem);
