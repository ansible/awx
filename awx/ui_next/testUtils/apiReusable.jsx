// eslint-disable-next-line import/prefer-default-export
export function describeNotificationMixin (Model, name) {
  describe(name, () => {
    const mockHttp = ({ post: jest.fn(() => Promise.resolve()) });
    const ModelAPI = new Model(mockHttp);

    afterEach(() => {
      jest.clearAllMocks();
    });

    const parameters = ['success', 'error'];

    parameters.forEach((type) => {
      const label = `[notificationType=${type}, associationState=true`;
      const testName = `associateNotificationTemplate ${label} makes expected http calls`;

      test(testName, async (done) => {
        await ModelAPI.associateNotificationTemplate(1, 21, type);

        const expectedPath = `${ModelAPI.baseUrl}1/notification_templates_${type}/`;
        expect(mockHttp.post).toHaveBeenCalledTimes(1);

        const expectedParams = { id: 21 };
        expect(mockHttp.post.mock.calls.pop()).toEqual([expectedPath, expectedParams]);

        done();
      });
    });

    parameters.forEach((type) => {
      const label = `[notificationType=${type}, associationState=false`;
      const testName = `disassociateNotificationTemplate ${label} makes expected http calls`;

      test(testName, async (done) => {
        await ModelAPI.disassociateNotificationTemplate(1, 21, type);

        const expectedPath = `${ModelAPI.baseUrl}1/notification_templates_${type}/`;
        expect(mockHttp.post).toHaveBeenCalledTimes(1);

        const expectedParams = { id: 21, disassociate: true };
        expect(mockHttp.post.mock.calls.pop()).toEqual([expectedPath, expectedParams]);

        done();
      });
    });
  });
}
