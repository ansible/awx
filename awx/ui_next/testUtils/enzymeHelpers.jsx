/*
 * Enzyme helpers for injecting top-level contexts
 * derived from https://lingui.js.org/guides/testing.html
 */
import React from 'react';
import { shape, object, string, arrayOf } from 'prop-types';
import { mount, shallow } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';
import { I18nProvider } from '@lingui/react';
import { ConfigProvider } from '../src/contexts/Config';

const language = 'en-US';
const intlProvider = new I18nProvider(
  {
    language,
    catalogs: {
      [language]: {},
    },
  },
  {}
);
const {
  linguiPublisher: { i18n: originalI18n },
} = intlProvider.getChildContext();

const defaultContexts = {
  linguiPublisher: {
    i18n: {
      ...originalI18n,
      _: key => key.id, // provide _ macro, for just passing down the key
      toJSON: () => '/i18n/',
    },
  },
  config: {
    ansible_version: null,
    custom_virtualenvs: [],
    version: null,
    toJSON: () => '/config/',
  },
  router: {
    history_: {
      push: () => {},
      replace: () => {},
      createHref: () => {},
      listen: () => {},
      location: {
        hash: '',
        pathname: '',
        search: '',
        state: '',
      },
      toJSON: () => '/history/',
    },
    route: {
      location: {
        hash: '',
        pathname: '',
        search: '',
        state: '',
      },
      match: {
        params: {},
        isExact: false,
        path: '',
        url: '',
      },
    },
    toJSON: () => '/router/',
  },
};

function wrapContexts(node, context) {
  const { config, router } = context;
  class Wrap extends React.Component {
    render() {
      // eslint-disable-next-line react/no-this-in-sfc
      const { children, ...props } = this.props;
      const component = React.cloneElement(children, props);
      if (router.history) {
        return (
          <ConfigProvider value={config}>
            <Router history={router.history}>{component}</Router>
          </ConfigProvider>
        );
      }
      return (
        <ConfigProvider value={config}>
          <MemoryRouter>{component}</MemoryRouter>
        </ConfigProvider>
      );
    }
  }

  return <Wrap>{node}</Wrap>;
}

function applyDefaultContexts(context) {
  if (!context) {
    return defaultContexts;
  }
  const newContext = {};
  Object.keys(defaultContexts).forEach(key => {
    newContext[key] = {
      ...defaultContexts[key],
      ...context[key],
    };
  });
  return newContext;
}

export function shallowWithContexts(node, options = {}) {
  const context = applyDefaultContexts(options.context);
  return shallow(wrapContexts(node, context));
}

export function mountWithContexts(node, options = {}) {
  const context = applyDefaultContexts(options.context);
  const childContextTypes = {
    linguiPublisher: shape({
      i18n: object.isRequired,
    }).isRequired,
    config: shape({
      ansible_version: string,
      custom_virtualenvs: arrayOf(string),
      version: string,
    }),
    router: shape({
      route: shape({
        location: shape({}),
        match: shape({}),
      }).isRequired,
      history: shape({}),
    }),
    ...options.childContextTypes,
  };
  return mount(wrapContexts(node, context), { context, childContextTypes });
}

/**
 * Wait for element(s) to achieve a desired state.
 *
 * @param[wrapper] - A ReactWrapper instance
 * @param[selector] - The selector of the element(s) to wait for.
 * @param[callback] - Callback to poll - by default this checks for a node count of 1.
 */
export function waitForElement(
  wrapper,
  selector,
  callback = el => el.length === 1
) {
  const interval = 100;
  return new Promise((resolve, reject) => {
    let attempts = 30;
    (function pollElement() {
      wrapper.update();
      const el = wrapper.find(selector);
      if (callback(el)) {
        return resolve(el);
      }
      if (--attempts <= 0) {
        const message = `Expected condition for <${selector}> not met: ${callback.toString()}`;
        return reject(new Error(message));
      }
      return setTimeout(pollElement, interval);
    })();
  });
}
