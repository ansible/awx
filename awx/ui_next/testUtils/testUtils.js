
const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));
/* eslint-disable-next-line import/prefer-default-export */
export { sleep };
