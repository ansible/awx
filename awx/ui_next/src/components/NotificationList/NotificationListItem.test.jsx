import React from 'react';
import { mountWithContexts } from '../../../testUtils/enzymeHelpers';
import NotificationListItem from './NotificationListItem';

describe('<NotificationListItem canToggleNotifications />', () => {
  let wrapper;
  let toggleNotification;

  const mockNotif = {
    id: 9000,
    name: 'Foo',
    notification_type: 'slack',
  };

  const typeLabels = {
    slack: 'Slack',
  };

  beforeEach(() => {
    toggleNotification = jest.fn();
  });

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
      wrapper = null;
    }
    jest.clearAllMocks();
  });

  test('initially renders succesfully and displays correct label', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    expect(wrapper.find('NotificationListItem')).toMatchSnapshot();
  });

  test('displays correct label in correct column', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    const typeCell = wrapper
      .find('DataListCell')
      .at(1)
      .find('div');
    expect(typeCell.text()).toContain('Slack');
  });

  test('handles start click when toggle is on', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        startedTurnedOn
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    wrapper
      .find('Switch')
      .first()
      .find('input')
      .simulate('change');
    expect(toggleNotification).toHaveBeenCalledWith(9000, true, 'started');
  });

  test('handles start click when toggle is off', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        startedTurnedOn={false}
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    wrapper
      .find('Switch[aria-label="Toggle notification start"]')
      .find('input')
      .simulate('change');
    expect(toggleNotification).toHaveBeenCalledWith(9000, false, 'started');
  });

  test('handles error click when toggle is on', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        successTurnedOn
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    wrapper
      .find('Switch[aria-label="Toggle notification success"]')
      .find('input')
      .simulate('change');
    expect(toggleNotification).toHaveBeenCalledWith(9000, true, 'success');
  });

  test('handles error click when toggle is off', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        successTurnedOn={false}
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    wrapper
      .find('Switch[aria-label="Toggle notification success"]')
      .find('input')
      .simulate('change');
    expect(toggleNotification).toHaveBeenCalledWith(9000, false, 'success');
  });

  test('handles error click when toggle is on', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        errorTurnedOn
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    wrapper
      .find('Switch[aria-label="Toggle notification failure"]')
      .find('input')
      .simulate('change');
    expect(toggleNotification).toHaveBeenCalledWith(9000, true, 'error');
  });

  test('handles error click when toggle is off', () => {
    wrapper = mountWithContexts(
      <NotificationListItem
        notification={mockNotif}
        errorTurnedOn={false}
        toggleNotification={toggleNotification}
        detailUrl="/foo"
        canToggleNotifications
        typeLabels={typeLabels}
      />
    );
    wrapper
      .find('Switch[aria-label="Toggle notification failure"]')
      .find('input')
      .simulate('change');
    expect(toggleNotification).toHaveBeenCalledWith(9000, false, 'error');
  });
});
