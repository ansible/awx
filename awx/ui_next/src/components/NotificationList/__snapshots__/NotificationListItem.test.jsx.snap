// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationListItem canToggleNotifications /> initially renders succesfully and displays correct label 1`] = `
<NotificationListItem
  approvalsTurnedOn={false}
  canToggleNotifications={true}
  detailUrl="/foo"
  errorTurnedOn={false}
  i18n={"/i18n/"}
  notification={
    Object {
      "id": 9000,
      "name": "Foo",
      "notification_type": "slack",
    }
  }
  showApprovalsToggle={false}
  startedTurnedOn={false}
  successTurnedOn={false}
  toggleNotification={[MockFunction]}
  typeLabels={
    Object {
      "slack": "Slack",
    }
  }
>
  <Tr
    id="notification-row-9000"
  >
    <TrBase
      id="notification-row-9000"
      innerRef={null}
    >
      <tr
        className=""
        data-ouia-component-id="OUIA-Generated-TableRow-1"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe={true}
        hidden={false}
        id="notification-row-9000"
      >
        <Td
          dataLabel="Name"
          id="notification-9000"
        >
          <TdBase
            dataLabel="Name"
            id="notification-9000"
            innerRef={null}
          >
            <td
              className=""
              data-label="Name"
              id="notification-9000"
            >
              <Link
                to="/foo"
              >
                <LinkAnchor
                  href="/foo"
                  navigate={[Function]}
                >
                  <a
                    href="/foo"
                    onClick={[Function]}
                  >
                    <b>
                      Foo
                    </b>
                  </a>
                </LinkAnchor>
              </Link>
            </td>
          </TdBase>
        </Td>
        <Td
          dataLabel="Type"
        >
          <TdBase
            dataLabel="Type"
            innerRef={null}
          >
            <td
              className=""
              data-label="Type"
            >
              Slack
            </td>
          </TdBase>
        </Td>
        <ActionsTd
          dataLabel="Options"
          gridColumns="120px 120px 120px 120px"
        >
          <ActionsTd___StyledTd
            _css={160}
            dataLabel="Options"
          >
            <StyledComponent
              _css={160}
              dataLabel="Options"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ActionsTd___StyledTd-sc-1ys3lw-1",
                    "isStatic": false,
                    "lastClassName": "dIDjZI",
                    "rules": Array [
                      "text-align:right;--pf-c-table--cell--Width:",
                      [Function],
                      "px;",
                    ],
                  },
                  "displayName": "ActionsTd___StyledTd",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "ActionsTd___StyledTd-sc-1ys3lw-1",
                  "target": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Td",
                    "render": [Function],
                  },
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <Td
                _css={160}
                className="ActionsTd___StyledTd-sc-1ys3lw-1 dIDjZI"
                dataLabel="Options"
              >
                <TdBase
                  _css={160}
                  className="ActionsTd___StyledTd-sc-1ys3lw-1 dIDjZI"
                  dataLabel="Options"
                  innerRef={null}
                >
                  <td
                    _css={160}
                    className="ActionsTd___StyledTd-sc-1ys3lw-1 dIDjZI"
                    data-label="Options"
                  >
                    <ActionsGrid
                      gridColumns="120px 120px 120px 120px"
                      numActions={4}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "ActionsTd__ActionsGrid-sc-1ys3lw-0",
                              "isStatic": false,
                              "lastClassName": "dkSSIN",
                              "rules": Array [
                                "display:grid;grid-gap:16px;align-items:center;",
                                [Function],
                              ],
                            },
                            "displayName": "ActionsGrid",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "ActionsTd__ActionsGrid-sc-1ys3lw-0",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        gridColumns="120px 120px 120px 120px"
                        numActions={4}
                      >
                        <div
                          className="ActionsTd__ActionsGrid-sc-1ys3lw-0 dkSSIN"
                        >
                          <ActionItem
                            column={1}
                            key=".0"
                            visible={false}
                          />
                          <ActionItem
                            column={2}
                            key=".1"
                            visible={true}
                          >
                            <ActionItem___StyledDiv
                              _css={2}
                            >
                              <StyledComponent
                                _css={2}
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "ActionItem___StyledDiv-sc-1x1z9nz-0",
                                      "isStatic": false,
                                      "lastClassName": "gydHvj",
                                      "rules": Array [
                                        "grid-column:",
                                        [Function],
                                        ";",
                                      ],
                                    },
                                    "displayName": "ActionItem___StyledDiv",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "ActionItem___StyledDiv-sc-1x1z9nz-0",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="ActionItem___StyledDiv-sc-1x1z9nz-0 cSmxUZ"
                                >
                                  <Switch
                                    aria-label="Toggle notification start"
                                    id="notification-9000-started-toggle"
                                    isChecked={false}
                                    isDisabled={false}
                                    label="Start"
                                    labelOff="Start"
                                    onChange={[Function]}
                                  >
                                    <label
                                      className="pf-c-switch"
                                      data-ouia-component-id="OUIA-Generated-Switch-1"
                                      data-ouia-component-type="PF4/Switch"
                                      data-ouia-safe={true}
                                      htmlFor="notification-9000-started-toggle"
                                    >
                                      <input
                                        aria-label="Toggle notification start"
                                        aria-labelledby={null}
                                        checked={false}
                                        className="pf-c-switch__input"
                                        disabled={false}
                                        id="notification-9000-started-toggle"
                                        onChange={[Function]}
                                        type="checkbox"
                                      />
                                      <span
                                        className="pf-c-switch__toggle"
                                      />
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-switch__label pf-m-on"
                                        id={null}
                                      >
                                        Start
                                      </span>
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-switch__label pf-m-off"
                                        id={null}
                                      >
                                        Start
                                      </span>
                                    </label>
                                  </Switch>
                                </div>
                              </StyledComponent>
                            </ActionItem___StyledDiv>
                          </ActionItem>
                          <ActionItem
                            column={3}
                            key=".2"
                            visible={true}
                          >
                            <ActionItem___StyledDiv
                              _css={3}
                            >
                              <StyledComponent
                                _css={3}
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "ActionItem___StyledDiv-sc-1x1z9nz-0",
                                      "isStatic": false,
                                      "lastClassName": "gydHvj",
                                      "rules": Array [
                                        "grid-column:",
                                        [Function],
                                        ";",
                                      ],
                                    },
                                    "displayName": "ActionItem___StyledDiv",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "ActionItem___StyledDiv-sc-1x1z9nz-0",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="ActionItem___StyledDiv-sc-1x1z9nz-0 fZyHIW"
                                >
                                  <Switch
                                    aria-label="Toggle notification success"
                                    id="notification-9000-success-toggle"
                                    isChecked={false}
                                    isDisabled={false}
                                    label="Success"
                                    labelOff="Success"
                                    onChange={[Function]}
                                  >
                                    <label
                                      className="pf-c-switch"
                                      data-ouia-component-id="OUIA-Generated-Switch-2"
                                      data-ouia-component-type="PF4/Switch"
                                      data-ouia-safe={true}
                                      htmlFor="notification-9000-success-toggle"
                                    >
                                      <input
                                        aria-label="Toggle notification success"
                                        aria-labelledby={null}
                                        checked={false}
                                        className="pf-c-switch__input"
                                        disabled={false}
                                        id="notification-9000-success-toggle"
                                        onChange={[Function]}
                                        type="checkbox"
                                      />
                                      <span
                                        className="pf-c-switch__toggle"
                                      />
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-switch__label pf-m-on"
                                        id={null}
                                      >
                                        Success
                                      </span>
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-switch__label pf-m-off"
                                        id={null}
                                      >
                                        Success
                                      </span>
                                    </label>
                                  </Switch>
                                </div>
                              </StyledComponent>
                            </ActionItem___StyledDiv>
                          </ActionItem>
                          <ActionItem
                            column={4}
                            key=".3"
                            visible={true}
                          >
                            <ActionItem___StyledDiv
                              _css={4}
                            >
                              <StyledComponent
                                _css={4}
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "ActionItem___StyledDiv-sc-1x1z9nz-0",
                                      "isStatic": false,
                                      "lastClassName": "gydHvj",
                                      "rules": Array [
                                        "grid-column:",
                                        [Function],
                                        ";",
                                      ],
                                    },
                                    "displayName": "ActionItem___StyledDiv",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "ActionItem___StyledDiv-sc-1x1z9nz-0",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="ActionItem___StyledDiv-sc-1x1z9nz-0 gydHvj"
                                >
                                  <Switch
                                    aria-label="Toggle notification failure"
                                    id="notification-9000-error-toggle"
                                    isChecked={false}
                                    isDisabled={false}
                                    label="Failure"
                                    labelOff="Failure"
                                    onChange={[Function]}
                                  >
                                    <label
                                      className="pf-c-switch"
                                      data-ouia-component-id="OUIA-Generated-Switch-3"
                                      data-ouia-component-type="PF4/Switch"
                                      data-ouia-safe={true}
                                      htmlFor="notification-9000-error-toggle"
                                    >
                                      <input
                                        aria-label="Toggle notification failure"
                                        aria-labelledby={null}
                                        checked={false}
                                        className="pf-c-switch__input"
                                        disabled={false}
                                        id="notification-9000-error-toggle"
                                        onChange={[Function]}
                                        type="checkbox"
                                      />
                                      <span
                                        className="pf-c-switch__toggle"
                                      />
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-switch__label pf-m-on"
                                        id={null}
                                      >
                                        Failure
                                      </span>
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-switch__label pf-m-off"
                                        id={null}
                                      >
                                        Failure
                                      </span>
                                    </label>
                                  </Switch>
                                </div>
                              </StyledComponent>
                            </ActionItem___StyledDiv>
                          </ActionItem>
                        </div>
                      </StyledComponent>
                    </ActionsGrid>
                  </td>
                </TdBase>
              </Td>
            </StyledComponent>
          </ActionsTd___StyledTd>
        </ActionsTd>
      </tr>
    </TrBase>
  </Tr>
</NotificationListItem>
`;
